freebsd-stubs.h

```diff
@@ -0,0 +1,14 @@
+#ifndef _FREEBSD_STUBS_H
+#define _FREEBSD_STUBS_H
+
+#define O_PATH 0
```

Comment:
Interesting note: https://github.com/flatpak/xdg-desktop-portal/blob/cb54478e07f1b51537ee75179863c02b5ef97c01/src/xdp-utils.c#L937-L941 - seems like regular fds are preferred now.

So would be better to `#ifdef O_PATH` all the support for `O_PATH`

Reply:
There are a lot of places where it is used. For instance, `document-portal/document-portal-fuse.c:909`:

```
static int
xdp_nonphysical_document_inode_opendir (XdpInode *inode)
{
  XdpDomain *domain = inode->domain;
  xdp_autofd int dirfd = -1;
  int res;

  g_assert (domain->type == XDP_DOMAIN_DOCUMENT);
  g_assert (inode->physical == NULL);

  dirfd = open (domain->doc_path, O_PATH | O_DIRECTORY);
  if (dirfd < 0)
  ...
```

ifdef'ing the `open` call would obviously break this code. Or did I misunderstood you?

Reply:
Yes, in that file, we can only define `O_PATH` to 0, but in the linked `xdp-utils.c` we can disable a whole section handling it.

(again, not sure how useful the document portal would be really, maybe just drop it)